stetho/src/main/java/com/facebook/stetho/inspector/network/ResourceTypeHelper.java

```diff
@@ -24,9 +24,18 @@ public ResourceTypeHelper() {
     // know exactly why, but whatever.
     mMimeMatcher.addRule("text/javascript", Page.ResourceType.XHR);
     mMimeMatcher.addRule("application/json", Page.ResourceType.XHR);
+    mMimeMatcher.addRule(
+        DownloadingAsyncPrettyPrinterFactory.PrettyPrinterDisplayType.JSON.getDisplayType(),
+        Page.ResourceType.XHR);
 
     // Everything else gets a lame, unformatted blob.
     mMimeMatcher.addRule("text/*", Page.ResourceType.DOCUMENT);
+    mMimeMatcher.addRule(
+        DownloadingAsyncPrettyPrinterFactory.PrettyPrinterDisplayType.HTML.getDisplayType(),
```

Comment:
_VERY_ bad collusion between unrelated types!  You really must be careful about how you mix abstractions so that folks aren't surprised by two unrelated layers suddenly mixing together. 

In this case, you want an adapter that simply knows how to convert `PrettyPrinterDisplayType` values to `Page.ResourceType`.  `ResourceTypeHelper` was explicitly designed to convert MIME types to `Page.ResourceType`.  I'd recommend putting this adapter in `PrettyPrinterDisplayType` itself, with a `getResourceType` method instead of your `getDisplayType` method today.